### PR TITLE
Handle Errors when Downstream Steps have Failed But Upstream is Not Blocked

### PR DESCRIPTION
**Pull request by angelosantos4**

### CODE REVIEW

File: nodestream/pipeline/pipeline.py [hunk 1]
```diff
@@ -144,6 +144,10 @@ def pipeline_has_died(self) -> bool:
 
     async def submit_object_or_die_trying(self, obj):
         should_continue = True
+        if self.pipeline_has_died():
```
Comment (zprobst) [inline nodestream/pipeline/pipeline.py:147]: I was expecting this to be in the loop before the await. What was the logic behind putting this in the head?

**Review by zprobst — APPROVED** [review]

**Review by zprobst — CHANGES_REQUESTED** [review]
Looks like these tests are flaky. Can you take a look?

**Review by ccloes — APPROVED** [review]
Reviewed as a team during standup, waiting for Zach's review

**Review by ccloes — APPROVED** [review]
Still waiting on Zach's review.

**Review by zprobst — APPROVED** [review]

**Pull request merged**